graphql/directive_test.go

```diff
@@ -891,3 +891,78 @@ func TestRequirePatchOwner(t *testing.T) {
 		})
 	}
 }
+func TestRequireAdmin(t *testing.T) {
+	defer func() {
+		err := db.ClearCollections(user.Collection)
+		require.NoError(t, err, "unable to clear user collection")
+	}()
+
+	obj := any(nil)
+
+	for tName, tCase := range map[string]func(ctx context.Context, t *testing.T, config Config, usr *user.DBUser){
+		"SucceedsWhenUserIsAdmin": func(ctx context.Context, t *testing.T, config Config, usr *user.DBUser) {
+			role := "superuser"
+			require.NoError(t, usr.AddRole(ctx, role), "Failed to add role")
+			defer func() {
+				require.NoError(t, usr.RemoveRole(ctx, role), "Failed to remove role")
+			}()
+
+			nextCalled := false
+			wrappedNext := func(rctx context.Context) (any, error) {
+				nextCalled = true
+				return nil, nil
+			}
+
+			res, err := config.Directives.RequireAdmin(ctx, obj, wrappedNext)
+			assert.NoError(t, err, "RequireAdmin failed unexpectedly for admin user")
+			assert.Nil(t, res, "Result should be nil when permissions succeed")
+			assert.True(t, nextCalled, "next middleware should be called")
+		},
+		"FailsWhenUserIsNotAdmin": func(ctx context.Context, t *testing.T, config Config, usr *user.DBUser) {
+			role := "regularuser"
+			require.NoError(t, usr.AddRole(ctx, role), "Failed to add role")
+			defer func() {
+				require.NoError(t, usr.RemoveRole(ctx, role), "Failed to remove role")
+			}()
+
+			nextCalled := false
+			wrappedNext := func(rctx context.Context) (any, error) {
+				nextCalled = true
+				return nil, nil
+			}
+
+			res, err := config.Directives.RequireAdmin(ctx, obj, wrappedNext)
+			assert.EqualError(t, err, "input: User 'Evergreen User' lacks required permissions", "Expected permission error")
+			assert.Nil(t, res, "Result should be nil when permissions fail")
+			assert.False(t, nextCalled, "next middleware should not be called")
+		},
+		"FailsWhenUserHasNoRoles": func(ctx context.Context, t *testing.T, config Config, usr *user.DBUser) {
+			nextCalled := false
+			wrappedNext := func(rctx context.Context) (any, error) {
+				nextCalled = true
+				return nil, nil
+			}
+
+			res, err := config.Directives.RequireAdmin(ctx, obj, wrappedNext)
+			assert.EqualError(t, err, "input: User 'Evergreen User' lacks required permissions", "Expected permission error")
+			assert.Nil(t, res, "Result should be nil when permissions fail")
+			assert.False(t, nextCalled, "next middleware should not be called")
```

Comment:
nit. while I think these error messages are helpful, I think we should remove them as they don't match with other tests in this repo
```suggestion
			assert.Nil(t, res)
			assert.False(t, nextCalled)
```